packages/react-dom/src/__tests__/ReactDOMComponent-test.js

```diff
@@ -1151,19 +1154,37 @@ describe('ReactDOMComponent', () => {
         <input type="text" defaultValue="foo" />,
         container,
       );
+      var getTrackedValue = getValueTracker(inst);
+      expect(getTrackedValue.call(inst)).toEqual('foo');
 
-      var tracker = inputValueTracking._getTrackerFromNode(inst);
+      inst.defaultValue = 'new foo';
+      expect(getTrackedValue.call(inst)).not.toEqual('new foo');
+      expect(getTrackedValue.call(inst)).toEqual('foo');
 
-      expect(tracker.getValue()).toEqual('foo');
+      inst.value = 'bar';
+      expect(getTrackedValue.call(inst)).toEqual('bar');
+
+      // even if the value changes to empty string, should not return to defaultValue
+      inst.value = '';
+      expect(getTrackedValue.call(inst)).toEqual('');
     });
 
     it('should track textarea values', () => {
       var container = document.createElement('div');
       var inst = ReactDOM.render(<textarea defaultValue="foo" />, container);
+      var getTrackedValue = getValueTracker(inst);
+      expect(getTrackedValue.call(inst)).toEqual('foo');
+
+      inst.defaultValue = 'new foo';
+      expect(getTrackedValue.call(inst)).not.toEqual('new foo');
+      expect(getTrackedValue.call(inst)).toEqual('foo');
 
-      var tracker = inputValueTracking._getTrackerFromNode(inst);
+      inst.value = 'bar';
+      expect(getTrackedValue.call(inst)).toEqual('bar');
 
-      expect(tracker.getValue()).toEqual('foo');
+      // even if the value changes to empty string, should not return to defaultValue
+      inst.value = '';
```

Comment:
Does the test still pass if you set the tracker instead of directly setting inst value?